broker/src/main/java/io/zeebe/broker/system/partitions/impl/AtomixPartitionMessagingService.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker.system.partitions.impl;
+
+import io.atomix.cluster.ClusterMembershipService;
+import io.atomix.cluster.Member;
+import io.atomix.cluster.MemberId;
+import io.atomix.cluster.messaging.ClusterCommunicationService;
+import io.zeebe.broker.system.partitions.PartitionMessagingService;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.Executor;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public class AtomixPartitionMessagingService implements PartitionMessagingService {
+  private final ClusterCommunicationService communicationService;
+  private final ClusterMembershipService clusterMembershipService;
+  private final Set<MemberId> otherMembers;
+
+  public AtomixPartitionMessagingService(
+      final ClusterCommunicationService communicationService,
+      final ClusterMembershipService clusterMembershipService,
+      final Collection<MemberId> members) {
+    this.communicationService = communicationService;
+    this.clusterMembershipService = clusterMembershipService;
+    this.otherMembers = getOtherMemberIds(clusterMembershipService, members);
+  }
+
+  @Override
+  public void subscribe(
+      final String subject, final Consumer<ByteBuffer> consumer, final Executor executor) {
+    communicationService.subscribe(subject, consumer, executor);
+  }
+
+  @Override
+  public void broadcast(final String subject, final ByteBuffer payload) {
+    final var reachableMembers =
+        otherMembers.stream().filter(this::isReachable).collect(Collectors.toUnmodifiableSet());
+
+    communicationService.multicast(subject, payload, reachableMembers);
+  }
+
+  @Override
+  public void unsubscribe(final String subject) {
+    communicationService.unsubscribe(subject);
+  }
+
+  private Set<MemberId> getOtherMemberIds(
+      final ClusterMembershipService clusterMembershipService, final Collection<MemberId> members) {
+    final var localMemberId = clusterMembershipService.getLocalMember().id();
```

Comment:
This only works when all nodes configured as initial entries right?

Reply:
Sort of - it assumes that the set of members in the Raft is static, so it's really just filtering out the local member. But this is independent of the discovery protocol.

Reply:
You use the ClusterMemberShipService to get the members or?

Reply:
It's used initially to get the local member ID, and later to check if a member is reachable. So this assumes member IDs are static, and the Raft configuration is also static - but whether they are reachable or not is dynamic and relies on SWIM.